### PR TITLE
docs(aws): revamp aws docs + iam role anywhere profile

### LINKED ISSUE #4130: Authenticate components to services using Dapr SPIFFE identity X.509 certificate
**What content needs to be created or modified?**
May need an explanation of this, and may need to update component specs to reflect this option

**Where should the new material be placed?**
[Components concept](https://docs.dapr.io/concepts/components-concept/)
[Component specs?](https://docs.dapr.io/reference/components-reference/)

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**
https://github.com/dapr/proposals/pull/51/files

### PR DESCRIPTION
**Pull request by sicoyle**
**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Bring changes from https://github.com/dapr/docs/pull/4371 to v1.15 docs.
- Revamp AWS docs given the complete refactor of things from the following contrib PRs: 
https://github.com/dapr/components-contrib/pull/3606
https://github.com/dapr/components-contrib/pull/3604
https://github.com/dapr/components-contrib/pull/3591
https://github.com/dapr/components-contrib/pull/3617

- Add new auth profiles for assuming an IAM role and for roles anywhere

## Issue reference

https://github.com/dapr/docs/issues/4130

### CODE REVIEW

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
Assume a specific IAM Role. This authentication profile assumes you're running the Dapr sidecar on EKS (AWS Kubernetes), or using a node/pod that has already been attached to an IAM policy defining access to AWS resources.
```

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: should we say 'in a future release' instead of promising Dapr 1.17?

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
Authentication can be done through setting [environment variables](https://docs.aws.amazon.com/sdkref/latest/guide/environment-variables.html) providing credentials. This supports situations such as running Dapr in self-hosted mode where the Dapr sidecar injector doesn't allow for configuring environment variables on the Dapr sidecar. 
```

**Comment by sicoyle** [comment]
is there a way i can view the results of my changes somewhere in the PR @hhunter-ms ?

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md:61]: what does ARN stand for? do we need to 'spell out' the acronym once somewhere in the doc?

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
Dapr components leveraging AWS services (for example, DynamoDB, SQS, S3) utilize standardized configuration attributes via the AWS SDK. [Learn more about how the AWS SDK handles credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials).
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review - thanks @sicoyle!

**Comment by hhunter-ms** [comment]
> is there a way i can view the results of my changes somewhere in the PR @hhunter-ms ?

@sicoyle there are a couple of ways to preview the docs site with your changes:
- open a PR from upstream instead of your fork. I did this for easier review of the [conversation api PR](https://github.com/dapr/docs/pull/4419)
- if you can't do that, you can build the docs site locally using [these instructions](https://github.com/dapr/docs?tab=readme-ov-file#run-local-server)

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (sicoyle) [reply daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: for deprecating things it is proper to deprecate after two releases is why I put 1.17.

**Review by hhunter-ms — APPROVED** [review]

**Comment by sicoyle** [comment]
do I need to do anything else to get this merged @hhunter-ms ?

**Comment by hhunter-ms** [comment]
> do I need to do anything else to get this merged @hhunter-ms ?

@sicoyle I'm just waiting for a quick review from @msfussell - should be merged today!

**Review by msfussell — APPROVED** [review]
LGTM

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (msfussell) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: Is this comment correct about Dapr v1.17? Do you mean v1.16 the next release?

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (msfussell) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: Same here

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (msfussell) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
| `accessKey` | N | AWS Access key id. Will be required in Dapr v1.17. | "AKIAIOSFODNN7EXAMPLE" |
```

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (msfussell) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
| `secretKey` | N | AWS Secret access key, used alongside `accessKey`. Will be required in Dapr v1.17. | "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" |
```

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (msfussell) [inline daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]:
```suggestion
| `assumeRoleArn` | N | ARN of the IAM role with AWS resource access. Will be required in Dapr v1.17. | "arn:aws:iam::123456789:role/mskRole" |
```

**Review by msfussell — COMMENTED** [review]
Left a few comments

File: daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md
Comment (hhunter-ms) [reply daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: @msfussell I left a comment about this too - here's @sicoyle's response: https://github.com/dapr/docs/pull/4456#discussion_r1870232821

Comment (sicoyle) [reply daprdocs/content/en/developing-applications/integrations/AWS/authenticating-aws.md]: yes, so it's 1.17 bc we're deprecating all aws prefixed fields. So, I've marked accessKey and secretKey as not required to help support the transition, so the docs are denoted and the metadata bundle as well to state that it will be required in dapr 1.17 as we rm the duplicate fields that had the aws prefix, and then the accessKey and secretKey fields can become required again.

**Comment by marcduiker** [comment]
@holopin-bot @sicoyle Thank you Sam! 🚀

**Pull request merged**
